python/src/MultigroupTable.python.cpp

```diff
@@ -0,0 +1,87 @@
+// system includes
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+// local includes
+#include "NDItk/MultigroupTable.hpp"
+#include "definitions.hpp"
+#include "read.hpp"
+
+// namespace aliases
+namespace python = pybind11;
+
+void wrapMultigroupTable( python::module& module, python::module& ) {
+
+  // type aliases
+  using Table = njoy::NDItk::MultigroupTable;
+  using Metadata = njoy::NDItk::multigroup::Metadata;
+  using Structure = njoy::NDItk::multigroup::Structure;
+  using FluxWeights = njoy::NDItk::multigroup::FluxWeights;
+  using ReactionCrossSections = njoy::NDItk::multigroup::ReactionCrossSections;
+
+  // wrap views created by this table
+
+  // create the table
+  python::class_< Table > table(
+
+    module,
+    "MultigroupTable",
+    "A multigroup neutron and photon table"
+  );
+
+  // wrap the table
+  table
+  .def(
+
+    python::init< std::string, std::string, std::string, std::string, 
+                  double, double, double, double,
+                  Structure, FluxWeights,
+                  ReactionCrossSections >(),
+    python::arg( "zaid" ), python::arg( "libname" ), python::arg( "source" ),
+    python::arg( "process" ), python::arg( "awr" ), python::arg( "weight" ),
+    python::arg( "temperature" ), python::arg( "dilution" ), 
+    python::arg( "structure" ), python::arg( "flux" ), python::arg( "xs" ),
+    "Initialise the table\n\n"
+    "Arguments:\n"
+    "    self           the table\n"
+    "    zaid           the zaid of the table\n"
+    "    libname        the library name\n"
+    "    source         the source date\n"
+    "    process        the processing date\n"
+    "    awr            the atomic weight ratio of the target (with respect\n"
+    "                   to the neutron mass)\n"
+    "    weight         the atomic weight of the target\n"
+    "    temperature    the temperature of the target\n"
+    "    dilution       the dilution (aka sigma0)\n"
+    "    structure      the primary group structure\n"
+    "    flux           the flux weights\n"
+    "    xs             the reaction cross section data"
+  )
+  .def_property_readonly(
+
+    "metadata",
+    &Table::metadata,
+    "The metadata of the table"
+  )
+  .def_property_readonly(
+
+    "structure",
+    &Table::structure,
+    "The primary group structure record"
+  )
+  .def_property_readonly(
+
+    "flux",
+    &Table::flux,
+    "The flux weight record"
+  )
+  .def_property_readonly(
+
+    "reaction_cross_sections",
+    &Table::reactionCrossSections,
+    "The reaction cross section record"
+  );
+
+  // add standard table definitions
```

Comment:
addStandardTableDefinitions doesn't seem to encapsulate everythin in NDItk/src/NDItk/MutigroupTable.hpp::MultigroupTable. The included read.hpp and the print don't seem to be part of this. They would need to be to be a full wrapper, yes?

Reply:
We only expose fromFile on full data tables, and I haven't added a toFile just yet to write one to a file. The read and print functions will not be exposed directly since these use iterators (and I also believe users shouldn't use these anyway).